File: validSudoku.java
import java.util.*;

class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashMap<Integer, Set<Character>> rows = new HashMap<>();
        HashMap<Integer, Set<Character>> cols = new HashMap<>();
        HashMap<String, Set<Character>> sqr = new HashMap<>();

        for(int r = 0; r<9; r++) {
            for(int c = 0; c<9; c++) {
                if (board[r][c] == '.') continue;

                String squareKey = (r/3) + "," + (c/3);

                if (rows.computeIfAbsent(r, k -> new HashSet<>()).contains(board[r][c])
                || cols.computeIfAbsent(c, k -> new HashSet<>()).contains(board[r][c])
                 || sqr.computeIfAbsent(squareKey, k -> new HashSet<>()).contains(board[r][c])) {
                    return false;
                }

                rows.get(r).add(board[r][c]);
                cols.get(c).add(board[r][c]);
                sqr.get(squareKey).add(board[r][c]);


            }
        }
        return true;
    }
}

File: ValidSudoku.go
func isValidSudoku(board [][]byte) bool {
	rows := make([]map[byte]bool, 9)
	cols := make([]map[byte]bool, 9)
	sqrs := make([]map[byte]bool, 9)
	for i := 0; i < 9; i++ {
		rows[i] = make(map[byte]bool)
		cols[i] = make(map[byte]bool)
		sqrs[i] = make(map[byte]bool)
	}

	for row, curRow := range board {
		for col, _ := range curRow {
			val := board[col][row]

			if val == '.' {
				continue
			}

			if rows[row][val] {
				return false
			}
			rows[row][val] = true

			if cols[col][val] {
				return false
			}
			cols[col][val] = true

			square := (row/3)*3 + col/3
			if sqrs[square][val] {
				return false
			}
			sqrs[square][val] = true
		}
	}
	return true
}